app/lib/cds_synchronizer.rb

```diff
@@ -0,0 +1,113 @@
+class CdsSynchronizer
+  extend TariffSynchronizer
+
+  # 1 - does not raise exception during record save
+  #   - logs cds error with xml node, record errors and exception
+  cattr_accessor :cds_logger_enabled
+  self.cds_logger_enabled = (ENV['TARIFF_CDS_LOGGER'].to_i == 1)
+
+  # set initial update date
+  # Initial dump date + 1 day
+  cattr_accessor :initial_update_date
+  self.initial_update_date = Date.new(2020, 9, 1)
+
+  class << self
+    def download
+      return Rails.logger.error 'Missing: Tariff sync enviroment variables: TARIFF_SYNC_USERNAME, TARIFF_SYNC_PASSWORD, TARIFF_SYNC_HOST and TARIFF_SYNC_EMAIL.' unless sync_variables_set?
+
+      TradeTariffBackend.with_redis_lock do
+        begin
+          TariffSynchronizer::CdsUpdate.sync(initial_date: initial_update_date)
+        rescue TariffUpdatesRequester::DownloadException => e
+          TariffLogger.failed_download(exception: e)
+          raise e.original
+        end
+        Rails.logger.info 'Finished downloading updates'
+      end
+    end
+
+    def apply
+      check_tariff_updates_failures
+      check_sequence
+
+      applied_updates = []
+      import_warnings = []
+
+      # The sync task is run on multiple machines to avoid more than on process
+      # running the apply task it is wrapped with a redis lock
+      TradeTariffBackend.with_redis_lock do
+        # Updates could be modifying primary keys so unrestricted it for all models.
+        sequel_models.each(&:unrestrict_primary_key)
+
+        subscribe 'apply.import_warnings' do |*args|
+          event = ActiveSupport::Notifications::Event.new(*args)
+          import_warnings << event.payload
+        end
+
+        date_range = date_range_since_oldest_pending_update
+        date_range.each do |day|
+          applied_updates << perform_update(CdsUpdate, day)
+        end
+
+        applied_updates.flatten!
+
+        if applied_updates.any? && BaseUpdate.pending_or_failed.none?
+          TariffLogger.apply(applied_updates.map(&:filename))
+
+          true
+        end
+      end
+    rescue Redlock::LockError
+      TariffLogger.apply_lock_error
```

Comment:
This still feels like an extra bit of misdirection just to log a line :thinking: